segmentation/utils/visualize.py

```diff
@@ -1,20 +1,152 @@
+    cell_data_stacked = cell_data_stacked.reindex(id_order, axis='index')
+    cell_data_stacked = cell_data_stacked.reindex(pop_order, axis='columns')
+
+    return cell_data_stacked
+
+
+def visualize_patient_population_distribution(cell_data, patient_col_name, population_col_name, color_map="jet",
+                                              show_total_count=True, show_distribution=True, show_proportion=True, save_dir=""):
+    """Plots the distribution of the population given by total count, direct count, and proportion
+              Args:
+                  cell_data: Pandas Dataframe containing columns with Patient ID and Cell Name
+                  patient_col_name: Name of column containing categorical Patient data
+                  population_col_name: Name of column in dataframe containing categorical Population data
+                  color_map: Name of MatPlotLib ColorMap used, default is jet
+                  show_total_count: Boolean specifying whether to show graph of total population count, default is true
+                  show_distribution: Boolean specifying whether to show graph of population distribution, default is true
+                  show_proportion: Boolean specifying whether to show graph of total count, default is true
+                  save_dir: Directory to save plots, default is blank"""
+    cell_data = cell_data.dropna()
+
+    # Plot by total count
+    if show_total_count:
+        population_values = cell_data[population_col_name].value_counts()
+        population_values.plot.bar(colormap=color_map)
+        plt.title("Distribution of Population in all patients")
+        plt.xlabel("Population Type")
+        plt.ylabel("Population Count")
+        plt.savefig(save_dir + "TotalPopulationDistribution.png")
+
+    # Plot by count
+    if show_distribution:
+        getSortedData(cell_data, patient_col_name, population_col_name).plot.bar(stacked=True, colormap=color_map)
+        plt.legend(loc='center left', bbox_to_anchor=(1.0, 0.5))
+        plt.xlabel("Patient ID")
+        plt.ylabel("Cell Count")
+        plt.title("Distribution of Population Count in Patients")
+        plt.savefig(save_dir + "PopulationDistribution.png")
+
+    # Plot by Proportion
+    if show_proportion:
+        getSortedData(cell_data, patient_col_name, population_col_name, is_normalized=True).plot.bar(stacked=True,
+                                                                                                     legend=False,
+                                                                                                     colormap=color_map)
+        plt.legend(loc='center left', bbox_to_anchor=(1.0, 0.5))
+        plt.xlabel("Patient ID")
+        plt.ylabel("Population Proportion")
+        plt.title("Distribution of Population Count Proportion in Patients")
+        plt.savefig(save_dir + "PopulationProportion.png")
+
```

Comment:
Yes, the idea of using a pull requests is that they're all separate from each other. I see that you merged in your plotting function branch. This is going to make things difficult to merge in. You should delete this branch and open a new PR with just the changes relevant for this functionality. This will make things simpler to review